src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -395,10 +395,19 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
         // This will dispose any existing (older kernels) associated with this notebook.
         // This way other parts of extension have access to this kernel immediately after event is handled.
         // Unlike webview notebooks we cannot revert to old kernel if kernel switching fails.
-        this.kernelProvider.getOrCreate(document.uri, {
+        const newKernel = this.kernelProvider.getOrCreate(document.uri, {
             metadata: selectedKernelConnectionMetadata
         });
 
+        // Auto start the local kernels.
+        if (
+            newKernel &&
+            !this.configuration.getSettings(undefined).disableJupyterAutoStart &&
+            newKernel.kernelConnectionMetadata.kind !== 'connectToLiveKernel'
```

Comment:
Is 'connectToLiveKernel' enough to indicate local? Don't remote kernels use 'connectToKernelSpec'?

Reply:
@rchiodo Pretty sure you are right on this. Remote as the option for connectToLiveKernel, but it doesn't have to be.

Reply:
Fixed